### PR TITLE
Drop privileges to snap_daemon user

### PR DESCRIPTION
**Pull request by jardon**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug Fix

* **What is the current behavior?** (You can also link to an open issue here)
mongod is run as `root` in the snap

* **What is the new behavior (if this is a feature change)?**
mongod is run as the `snap_daemon` user

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Nope.

* **Other information**:
[mongo docs for non-root users](https://www.mongodb.com/developer/products/mongodb/non-root-user-mongod-process/)
[snapcraft system-usernames docs](https://snapcraft.io/docs/system-usernames)

### CODE REVIEW

File: snap/local/start_mongod.sh [hunk 1]
```diff
@@ -6,6 +6,11 @@ if [ ! -f "${CONFIG_FILE}" ]; then
         echo "copying default config to ${CONFIG_FILE}"
```
Comment (MiaAltieri) [inline snap/local/start_mongod.sh:6]: nit: change the name of this folder to `snap/hooks/start_mongod.sh` for consistancy across DPE snaps

File: snap/local/start_mongod.sh [hunk 2]
```diff
@@ -6,6 +6,11 @@ if [ ! -f "${CONFIG_FILE}" ]; then
...
         cp -r $SNAP/etc/mongod.conf $SNAP_COMMON
 	mkdir -p $SNAP_COMMON/db
+	touch $SNAP_DATA/mongod.log
+
+        chown -R snap_daemon:root $SNAP_DATA
+        chown -R snap_daemon:root $SNAP_COMMON
 fi
 
-$SNAP/usr/bin/mongod --config $CONFIG_FILE "$@"
+$SNAP/usr/bin/setpriv --clear-groups --reuid snap_daemon \
```
Comment (MiaAltieri) [inline snap/local/start_mongod.sh:17]: nice 😎  -- might be useful to add a comment here like:
```
# For security measures, daemons should not be run as sudo. Execute mongod as the non-sudo user: snap-daemon.
```

**Review by MiaAltieri — APPROVED** [review]
LATM (looks amazing to me)

**Review by MiaAltieri — APPROVED** [review]

**Review by taurus-forever — APPROVED** [review]
Nice!

**Pull request merged**